server/api/crud/runs.py

```diff
@@ -468,3 +456,59 @@ async def _post_delete_run(project, uid):
                 project,
                 uid,
             )
+
+    @staticmethod
+    def _get_artifacts_from_uris(
+        db_session: sqlalchemy.orm.Session, project: str, run: dict
+    ):
+        """Fetch run artifacts by their artifact URIs in the run status"""
+        artifacts = []
+        artifact_uris = run.get("status", {}).get("artifact_uris", {})
+        for _, uri in artifact_uris.items():
+            _, uri = mlrun.datastore.parse_store_uri(uri)
+            project, key, iteration, tag, producer_id = mlrun.utils.parse_artifact_uri(
+                uri, project
+            )
+            artifact = server.api.crud.Artifacts().get_artifact(
+                db_session,
+                key=key,
+                tag=tag,
+                iter=iteration,
+                project=project,
+                producer_id=producer_id,
+                raise_on_not_found=False,
+            )
```

Comment:
so you have the key, tag, iteration, producer_id, etc. why not performing one API request to get all artifacts matching the query? performing single DB-call vs N-calls of GET is much better

Reply:
the producer id is the workflow id, I don't see how can I build a list query that ensures I get the correct artifacts